### PR TITLE
Fixes #54

### LINKED ISSUE #54: [BUG] Small memory leak when searching for device requirements
**Describe the bug**
Reported in YT comments. In `vulkan_device.c`, around line 322 the `requirements.device_exctension_names` darray is created, but not destroyed after the call to `physical_device_meets_requirements`. 

Instead of just adding this though, that whole block should be moved outside the loop, and the darray finally destroyed once the loop completes, before the check to see if a device was selected.

Edit: Also includes `required_validation_layer_names` and `available_layers` in `vulkan_backend.c` around line 242.

### PR DESCRIPTION
**Pull request by mikejzx**

### CODE REVIEW

**Comment by travisvroman** [comment]
Thanks for submitting this! If it's all the same to you, I'm going to wait on merging this until I record a video for a bugfix pass, and include it there.

File: engine/src/renderer/vulkan/vulkan_backend.c [hunk 1]
```diff
@@ -165,8 +171,11 @@ b8 vulkan_renderer_backend_initialize(renderer_backend* backend, const char* app
     VK_CHECK(vkCreateInstance(&create_info, context.allocator, &context.instance));
     KINFO("Vulkan Instance created.");
 
-    // Debugger
 #if defined(_DEBUG)
+    darray_destroy(required_validation_layer_names);
+    darray_destroy(required_extensions);
```
Comment (beewyka819) [inline engine/src/renderer/vulkan/vulkan_backend.c:176]: Shouldn't these two destroy calls be outside the _DEBUG block?

Comment (mikejzx) [reply engine/src/renderer/vulkan/vulkan_backend.c:176]: Ah, good catch -- the required_extensions destroy call needs to be moved out of that block.  The required_validation_layer_names one should remain there as it is created within another _DEBUG block.

**Review by travisvroman — CHANGES_REQUESTED** [review]
I know this is likely due to it being so long since I've looked at this, but there are merge conflicts. Would you mind resolving these conflicts and then I can approve/merge this?

Thank you!

**Comment by travisvroman** [comment]
Closing this due to inactivity.

**Pull request closed without merging**